subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/DependencyMetadataRulesIntegrationTest.groovy

```diff
@@ -221,19 +323,82 @@ abstract class DependencyMetadataRulesIntegrationTest extends AbstractHttpDepend
                 }
             }
         """
+        repositoryInteractions {
+            'org.test:moduleA:1.0' {
+                expectGetMetadata()
+                expectGetArtifact()
+            }
+            'org.test:moduleB:1.0'() {
+                expectGetMetadata()
+                expectGetArtifact()
+            }
+        }
 
         then:
         succeeds 'checkDep'
         def variantToTest = variantToTest
         resolve.expectGraph {
-            root(':', ':testproject:') {
+            root(':', ':test:') {
                 module("org.test:moduleA:1.0:$variantToTest") {
                     module('org.test:moduleB:1.0')
                 }
             }
         }
     }
 
+    @RequiredFeatures(
+        @RequiredFeature(feature = GradleMetadataResolveRunner.REPOSITORY_TYPE, value = "maven")
+    )
+    def "can set version on dependency constraint"() {
+        given:
+        repository {
+            'org.test:moduleA:1.0'() {
+                constraint 'org.test:moduleB:0.1'
+            }
+        }
+
+        when:
+        buildFile << """
+            dependencies {
+                $variantToTest 'org.test:moduleB'
+                components {
+                    withModule('org.test:moduleA') {
+                        withVariant("$variantToTest") { 
+                            withDependencyConstraints {
+                                it.each {
```

Comment:
The syntax is not very user friendly here. I know we deferred that, but at least it would be worth creating a ticket to make sure we don't forget about fixing the ugly `it.`. True for `it.each` and `it.version`.

Reply:
Agreed. Issue: https://github.com/gradle/gradle/issues/3731